src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -60,23 +59,25 @@ protected static RootExecutionNode BuildExecutionRootNode(ExecutionContext conte
 
             var fields = System.Threading.Interlocked.Exchange(ref context.ReusableFields, null) ?? new Fields();
 
-            SetSubFieldNodes(context, root, fields.CollectFrom(context, rootType, context.Operation.SelectionSet));
+            SetSubFieldNodes(context, root, fields.CollectFrom(context, rootType, context.Operation.SelectionSet, this));
 
             fields.Clear();
             System.Threading.Interlocked.CompareExchange(ref context.ReusableFields, fields, null);
 
             return root;
         }
 
+        bool IFieldCollectionStrategy.ShouldIncludeNode(ExecutionContext context, IHaveDirectives directives) => ExecutionHelper.ShouldIncludeNode(context, directives.Directives);
```

Comment:
I think IExecutionStrategy should inherit IFieldCollectionStrategy and this should simply be a protected virtual member of the class.

Reply:
And then add another IFieldCollectionStrategy.ShouldIncludeNode(c) => ShouldIncludeNode(c); so that it doesn’t need to be public

Reply:
I removed `IFieldCollectionStrategy `. It is not needed at all.